set.mm

```diff
@@ -788663,4 +788670,66 @@ to Davis and Putnam (1960).  (Contributed by David A. Wheeler,
 $( (End of Kunhao Zheng's mathbox.) $)
 $( End $[ set-mbox-kz.mm $] $)
 
+
+$( Begin $[ set-mbox-et.mm $] $)
+$( Skip $[ set-main.mm $] $)
+$(
+#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#
+  Mathbox for Ender Ting
+#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#
+$)
+
+
+$(
+=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+  Increasing sequences and subsequences
+=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+$)
+
+  $c UpWord $.
+
+  $( Extend class notation to include the set of strictly increasing
+     sequences. $)
+  cupword $a class UpWord S $.
+
+  ${
+    $d S k w $.
+
+    ${
+      $( Strictly increasing sequence is a sequence, adjacent elements of which
+         increase.  (Contributed by Ender Ting, 19-Nov-2024.) $)
+      df-upword $a |- UpWord S = { w | ( w e. Word S /\ A. k e. ( 0 ..^ ( ( # `
+      w ) - 1 ) ) ( w ` k ) < ( w ` ( k + 1 ) ) ) } $.
+    $}
+
+    ${
+      $( Empty set is an increasing sequence for every range.  (Contributed by
+         Ender Ting, 19-Nov-2024.) $)
+      upwordnul $p |- (/) e. UpWord S $=
+        ( vw vk c0 cv wcel cfv c1 co wbr cc0 chash cmin wceq wi cvv wb ax-mp cz
+        cle cword caddc clt cfzo wral wa cab cupword wal 0ex elab6g wrd0 eleq1a
+        fveq2 hash0 eqtrdi oveq1d 0red lem1d eqbrtrd eqeltrdi 1zzd zsubcld fzon
+        0z sylancr mpbid rzal syl jca mpgbir df-upword eleqtrri ) DBEZAUAZFZCEZ
+        VNGVQHUBIVNGUCJZCKVNLGZHMIZUDIZUEZUFZBUGZAUHDWDFZVNDNZWCOZBDPFWEWGBUIQU
+        JWCBDPUKRWFVPWBDVOFWFVPOAULDVOVNUMRWFWADNZWBWFVTKTJZWHWFVTKHMIKTWFVSKHM
+        WFVSDLGKVNDLUNUOUPZUQWFKWFURUSUTWFKSFVTSFWIWHQVEWFVSHWFVSKSWJVEVAWFVBVC
+        KVTVDVFVGVRCWAVHVIVJVKBACVLVM $.
+    $}
+
+    ${
+      $d A w $.
+
+      $( Any increasing sequence is a sequence.  (Contributed by Ender Ting,
+         19-Nov-2024.) $)
+      upwordisword $p |- ( A e. UpWord S -> A e. Word S ) $=
```

Comment:
(upwordisword) This could probably be shortened using ~ elrabi and ~ eleq2s if the definition was instead

```
      df-upword $a |- UpWord S = { w e. Word S |
        A. k e. ( 0 ..^ ( ( # ` w ) - 1 ) ) ( w ` k ) < ( w ` ( k + 1 ) ) } $.
```

(~ df-rab)

optional especially since upwordnul would have to be revised